core/controllers/base.py

```diff
@@ -274,35 +276,58 @@ def dispatch(self):
             self.redirect('/logout?redirect_url=%s' % self.request.uri)
             return
 
-        if self.payload is not None and self.REQUIRE_PAYLOAD_CSRF_CHECK:
```

Comment:
Initial approach with if-block -- If it is put or post request then validate csrf_token.
Current approach without if-block -- Check and validate csrf_token if it is present in self.request.get().
Current approach helps us to collect all of the args in dispatch method.

Regarding this please give me counter examples where things will not work, I have looked into the codebase but there is nothing stopping me to do this.